### PR TITLE
Combobox listbox refactor

### PR DESCRIPTION
**Pull request by nielsr98**
Added fixes based on feedback from Friday's test session.

### CODE REVIEW

File: src/cdk-experimental/combobox/combobox-panel.ts [hunk 1]
```diff
@@ -44,6 +44,10 @@ export class CdkComboboxPanel<T = unknown> {
 
   /** Registers the content's id and the content type with the panel. */
   _registerContent(contentId: string, contentType: AriaHasPopupValue) {
+    if (this.contentType && this.contentType !== contentType) {
+      return;
+    }
```
Comment (jelbourn) [inline src/cdk-experimental/combobox/combobox-panel.ts:49]: Add a comment that explains that changing the content type after opening isn't allowed?

File: src/cdk-experimental/combobox/combobox-popup.ts [hunk 2]
```diff
@@ -33,11 +41,21 @@ export class CdkComboboxPopup<T = unknown> implements OnInit {
...
   private _role: AriaHasPopupValue = 'dialog';
 
+  @Input()
+  get firstFocus(): HTMLElement {
+    return this._firstFocusElement;
+  }
+  set firstFocus(id: HTMLElement) {
+    this._firstFocusElement = id;
+  }
+  private _firstFocusElement: HTMLElement;
```
Comment (jelbourn) [inline src/cdk-experimental/combobox/combobox-popup.ts:51]: I don't quite follow this; if this directive is for dialog combobox content, I think the focus element would always be `this.elementRef.nativeElement`

File: src/cdk-experimental/combobox/combobox.ts [hunk 3]
```diff
@@ -74,9 +75,15 @@ export class CdkCombobox<T = unknown> implements OnDestroy, AfterContentInit {
   }
   private _openActions: OpenAction[] = ['click'];
 
+  @Input()
+  get autoSetText(): boolean { return this._autoSetText; }
+  set autoSetText(value: boolean) { this._autoSetText = coerceBooleanProperty(value); }
+  private _autoSetText: boolean = true;
```
Comment (jelbourn) [inline src/cdk-experimental/combobox/combobox.ts:81]: Add a JsDoc description that explains what this property does?

File: src/cdk-experimental/combobox/combobox.ts [hunk 4]
```diff
@@ -189,22 +211,29 @@ export class CdkCombobox<T = unknown> implements OnDestroy, AfterContentInit {
...
   }
 
-  private _setTextContent(content: T) {
+  private _setTextContent(content: T | T[]) {
     const contentArray = coerceArray(content);
     this._elementRef.nativeElement.textContent = contentArray.join(' ');
   }
 
+  private _isTextTrigger() {
+    const tagName = this._elementRef.nativeElement.tagName.toLowerCase();
```
Comment (jelbourn) [inline src/cdk-experimental/combobox/combobox.ts:233]: I would add a TODO to also test for `contenteditable` (I think supporting that will take some more thought before doing it)

File: src/cdk-experimental/listbox/listbox.ts [hunk 5]
```diff
@@ -355,7 +390,10 @@ export class CdkListbox<T> implements AfterContentInit, OnDestroy, OnInit, Contr
     }
 
     /** Will select an option if shift was pressed while navigating to the option */
-    const isArrow = (keyCode === UP_ARROW || keyCode === DOWN_ARROW);
+    const isArrow = (keyCode === UP_ARROW
+        || keyCode === DOWN_ARROW
+        || keyCode === LEFT_ARROW
+        || keyCode === RIGHT_ARROW);
```
Comment (jelbourn) [inline src/cdk-experimental/listbox/listbox.ts:396]: Should this always be all four, or should it depend on the orientation?

File: src/cdk-experimental/combobox/combobox-popup.ts (hunk 2 above)
Comment (nielsr98) [reply src/cdk-experimental/combobox/combobox-popup.ts:51]: My thought process was that if the popup is a dialog, then this.elementRef.nativeElement.focus() will just focus the entire popup. With this input, the user could choose to immediately focus some interior element like an input or listbox inside the dialog.

File: src/cdk-experimental/combobox/combobox.ts (hunk 3 above)
Comment (nielsr98) [reply src/cdk-experimental/combobox/combobox.ts:81]: Yeah will do.

File: src/cdk-experimental/combobox/combobox.ts (hunk 4 above)
Comment (nielsr98) [reply src/cdk-experimental/combobox/combobox.ts:233]: Sounds good.

File: src/cdk-experimental/listbox/listbox.ts (hunk 5 above)
Comment (nielsr98) [reply src/cdk-experimental/listbox/listbox.ts:396]: The isArrow variable is true if any arrow key is pressed. In the following if statement, a selection change will only occur if there is a new active option which the listKeyManager keeps track of. The listKeyManager knows the orientation, so if RIGHT is pressed while I vertical mode, the active item won't switch and even though isArrow is true no action is taken.

File: src/cdk-experimental/combobox/combobox-popup.ts (hunk 2 above)
Comment (jelbourn) [reply src/cdk-experimental/combobox/combobox-popup.ts:51]: We should be able to take care of that with the focus trap from cdk/a11y without having to add direct support for it to the pop-up here (in fact `CdkDialog` would take care of that automatically).

**Comment by jelbourn** [comment]
You can add merge-ready when ready

**Pull request merged**